### PR TITLE
Sync error code proto

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

`ErrorCode` in `mlflow/protos/databricks.proto` is out-of-date. This PR updates it.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@mlflow-automation autoformat

**Comment by harupy** [comment]
@mlflow-automation autoformat

File: mlflow/protos/databricks.proto [hunk 1]
```diff
@@ -185,30 +185,144 @@ message DocumentationMetadata {
...
-  // Internal, system-level error codes, which generally cannot be resolved by the user, but
-  // instead are due to service issues.
+  // It's assigned tag 0 to follow the best practice from
+  // https://developers.google.com/protocol-buffers/docs/style#enums
+  // 
+  // TODO(PLAT-55898): Add custom option to declare HTTP and gRPC mappings.
+  // Maps to:
+  // - google.rpc.Code: UNKNOWN = 2;
+  // - HTTP code: 500 Internal Server Error
+  UNKNOWN = 0;
```
Comment (harupy) [inline mlflow/protos/databricks.proto:198]: https://github.com/mlflow/mlflow/actions/runs/4602038407

```
mlflow/protos/model_registry.proto mlflow/protos/databricks_artifacts.proto mlflow/protos/mlflow_artifacts.proto mlflow/protos/scalapb/scalapb.proto
service.proto:435:3: "mlflow.UNKNOWN" is already defined in file "databricks.proto".
service.proto:435:3: Note that enum values use C++ scoping rules, meaning that enum values are siblings of their type, not children of it.  Therefore, "UNKNOWN" must be unique within "mlflow", not just within "SourceType".
```

`SourceType` enum also has a field name `UNKNOWN`.

File: mlflow/protos/databricks.proto [hunk 2]
```diff
@@ -185,30 +185,143 @@ message DocumentationMetadata {
...
   //
-  // Internal, system-level error codes, which generally cannot be resolved by the user, but
-  // instead are due to service issues.
+  // It's assigned tag 0 to follow the best practice from
+  // https://developers.google.com/protocol-buffers/docs/style#enums
+  // 
+  // Maps to:
+  // - google.rpc.Code: UNKNOWN = 2;
+  // - HTTP code: 500 Internal Server Error
+  // UNKNOWN = 0; commented out because this conflicts with `SourceType.UNKNOWN` in mlflow/protos/service.proto
```
Comment (dbczumar) [inline mlflow/protos/databricks.proto:197]: We can remove this :)

File: mlflow/protos/databricks.proto [hunk 3]
```diff
@@ -217,18 +330,82 @@ enum ErrorCode {
...
+
+  // Indicates that the HTTP response cannot be correctly deserialized.
+  // This currently is only used in DUST test clients, and not by any real service code.
+  UNPARSEABLE_HTTP_ERROR = 1009;
+
+  // The operation is not implemented or is not supported/enabled in this service.
+  //
+  // Maps to:
+  // - google.rpc.Code: UNIMPLEMENTED = 12;
+  // - HTTP code: 501 Not Implemented
```
Comment (dbczumar) [inline mlflow/protos/databricks.proto:392]: Does MlflowException map this to 501? Can we do that if not?

**Review by dbczumar — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/8158/files#r1156604029 is addressed. Thanks @harupy !

File: mlflow/protos/databricks.proto (hunk 3 above)
Comment (harupy) [reply mlflow/protos/databricks.proto:381]: Updated. We also need to map the new errors, right?

Comment (harupy) [reply mlflow/protos/databricks.proto:381]: Added `DATA_LOSS`.

**Pull request merged**